crates/sui-faucet/src/main.rs

```diff
@@ -82,6 +81,8 @@ async fn main() -> Result<(), anyhow::Error> {
         config,
     });
 
+    let batching_thread_app_state = Arc::clone(&app_state);
```

Comment:
This seems funky.  Why do we need it?

Reply:
I'm guessing this has something to do with why the `receiver` is protected by a `Mutex`.  We should try and architect this differently.  We can spawn the task that uses this when we create the faucet (after we create the channel), and give it the receiver side, rather than holding it in the app state in a way that requires us to add a `Mutex` that we should never use.